Character.cpp

```diff
@@ -98,20 +98,44 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+
+        if( *initialArmorLevel.get() > armor )
+        {
+            armor = *initialArmorLevel.get();
+        }
+
+        if( *initialAttackDamage.get() > attackDamage )
+        {
+            attackDamage = *initialAttackDamage.get();
+        }
+
+        if( *initialHitPoints.get() > hitPoints )
+        {
+            hitPoints = *initialHitPoints.get();
+        }
+
+        armor = armor * 1.1;
+        attackDamage = attackDamage * 1.1;
+        hitPoints = hitPoints * 1.1;
+
+        initialArmorLevel.reset(new int(armor));
+        initialAttackDamage.reset(new int(attackDamage));
+        initialHitPoints.reset(new int(hitPoints));
```

Comment:
- D.R.Y. refactor this.  you're doing the same thing 3 times with 6 variables. 
- Also, don't reset the unique_ptr<>'s.   Dereference them to update their held value.
- You don't need to dereference the thing returned from `.get()`.  dereferencing the unique_ptr will do that for you. 
- [ ] revise